android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -299,11 +312,10 @@ public static void setUsingTestConfiguration(boolean shouldUseTestConfiguration)
         testConfigurationFlag = shouldUseTestConfiguration;
     }
 
-    public void clearTestUpdates() {
-        if (isUsingTestConfiguration()) {
-            codePushPackage.clearTestUpdates();
-            removePendingUpdate();
-        }
+    public void clearUpdates() {
+        codePushPackage.clearUpdates();
```

Comment:
As I think about it, it doesn't seem too bad to add a new parameter to the constructor called "debugMode" or something. We could use it for this and we could also only clear the local cache file of that value is true.